spec/features/course_management_spec.rb

```diff
@@ -8,30 +8,28 @@
     let!(:user) { create(:user, role: :administrator) }
     before { login_as(user, scope: :user) }
 
-    describe 'course new page' do
-      before { visit new_course_path }
+    scenario 'Users can see a list of courses' do
+      create(:course, creator: user, updater: user)
 
-      it { is_expected.to have_field('course_title') }
-      it { is_expected.to have_field('course_description') }
+      visit courses_path
+      expect(all('.course').count).to eq(1)
+      expect(subject).to have_link(I18n.t('course.courses.index.new'), href: new_course_path)
     end
 
-    describe 'course creation' do
-      before { visit new_course_path }
-      subject { click_button I18n.t('helpers.submit.course.create') }
+    scenario 'Users can create a new course' do
```

Comment:
Actually, for the feature tests, shall we use branches? like:
```
feature 'User creates course'
  scenario 'with valid information' do
     xxx
  end

  scenario 'with valid title' do
     xxx
  end
end
```

Because it's a bit hard for me to read or write the tests without them, I also find some examples here: https://robots.thoughtbot.com/rspec-integration-tests-with-capybara